### ISSUE TITLE
Avoid reading whole file in memory

### ISSUE THREAD

**Issue by ckulkarni-cribl** [tjpoulose03/Cribl#1]
Hi Travis,

Thanks for submitting the take home! I noticed we're reading whole file in memory before returning a response. could we make that more efficient?

**Comment by tjpoulose03** [comment]
Yes, I believe I can  start reading the file from the bottom instead of the top and after getting each chunk parse it to check if the number of lines is met otherwise read the next chunk and so on, This way we don't need to read the entire file. I am working on the changes right now and will finish it by tomorrow evening if that is okay with you

**Comment by tjpoulose03** [comment]
I have made changes to the code so it does not read the entire file and only reads chunks starting from the bottom of the file.
